ibis/expr/types/relations.py

```diff
@@ -2481,6 +2484,314 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def __enter__(self):
         return self
 
+    def pivot_longer(
+        self,
+        cols: str | s.Selector,
+        *,
+        names_to: str | Iterable[str] = "name",
+        names_pattern: str | re.Pattern = r"(.+)",
+        names_transform: Callable[[str], ir.Value]
+        | Mapping[str, Callable[[str], ir.Value]]
+        | None = None,
+        values_to: str = "value",
+        values_transform: Callable[[ir.Value], ir.Value] | Deferred | None = None,
+    ) -> Table:
+        """Transform a table from wider to longer.
+
+        Parameters
+        ----------
+        cols
+            String column names or selectors.
+        names_to
+            A string or iterable of strings indicating how to name the new
+            pivoted columns.
+        names_pattern
+            Pattern to use to extract column names from the input. By default
+            the entire column name is extracted.
+        names_transform
+            Function or mapping of a name in `names_to` to a function to
+            transform a column name to a value.
+        values_to
+            Name of the pivoted value column.
+        values_transform
+            Apply a function to the value column. This can be a lambda or
+            deferred expression.
+
+        Returns
+        -------
+            Pivoted table
```

Comment:
Should this be indented like this? I'm used to the numpydoc style, not sure what our mkdocs api docs formats nicely.

Reply:
Ha! This is a rebase error on my part. Thanks for catching.

Reply:
Fixed!